cmd/provisioner-localpv/app/types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package app
+
+import (
+	mconfig "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+//Provisioner struct has the configuration and utilities required
+// across the different work-flows.
+type Provisioner struct {
+	stopCh      chan struct{}
+	kubeClient  *clientset.Clientset
+	namespace   string
+	helperImage string
+	// defaultConfig is the default configurations
+	// provided from ENV or Code
+	defaultConfig []mconfig.Config
+	// configParser is a reference to a function
+	configParser CASConfigParserFn
+}
+
+//CASConfigPVC struct contains the merged configuration of the PVC
+// and the associated SC. The configuration is derived from the
+// annotation `cas.openebs.io/config`. The configuration will be
+// in the following json format:
+// {
+//   Key1:{
+//	enabled: true
+//	value: "string value"
+//   },
+//   Key2:{
+//	enabled: true
+//	value: "string value"
+//   },
+// }
+type CASConfigPVC struct {
+	pvName  string
+	pvcName string
+	scName  string
+	config  map[string]interface{}
+}
+
+// CASConfigParserFn allows to provide pluggable parser function
+//  and makes it easy to unit test provisioner
+type CASConfigParserFn func(pvName string, pvc *v1.PersistentVolumeClaim) (*CASConfigPVC, error)
```

Comment:
This is great way to plug. :+1: 

Please check if we can remove the dependency of PVC

Reply:
Will check as part of the further refactoring.